modules/chanserv/main.c

```diff
@@ -255,165 +255,23 @@ c_ci_templates(mowgli_config_file_entry_t *ce)
-		ca2 = chanacs_find_host_by_user(mc, u, CA_AKICK);
-		if (ca2 != NULL)
-		{
-			if (chanban_find(chan, ca2->host, 'b') == NULL)
-			{
-				chanban_add(chan, ca2->host, 'b');
-				modestack_mode_param(chansvs.nick, chan, MTYPE_ADD, 'b', ca2->host);
-				modestack_flush_channel(chan);
-			}
-		}
-		else
-		{
-			// XXX this could be done more efficiently
-			ca2 = chanacs_find(mc, entity(u->myuser), CA_AKICK);
-			ban(chansvs.me->me, chan, u);
-		}
-		remove_ban_exceptions(chansvs.me->me, chan, u);
-		if (ca2 != NULL)
-		{
-			md = metadata_find(ca2, "reason");
-			if (md != NULL && *md->value != '|')
-			{
-				snprintf(akickreason, sizeof akickreason,
-						"Banned: %s", md->value);
-				p = strchr(akickreason, '|');
-				if (p != NULL)
-					*p = '\0';
-				else
-					p = akickreason + strlen(akickreason);
-				/* strip trailing spaces, so as not to
-				 * disclose the existence of an oper reason */
-				p--;
-				while (p > akickreason && *p == ' ')
-					p--;
-				p[1] = '\0';
-			}
-		}
-		try_kick(chansvs.me->me, chan, u, akickreason);
-		hdata->cu = NULL;
-		return;
-	}
-
-	/* Kick out users who may be recreating channels mlocked +i.
-	 * Users with +i flag are allowed to join, as are users matching
-	 * an invite exception (the latter only works if the channel already
-	 * exists because members are sent before invite exceptions).
-	 * Because most networks do not use kick_on_split_riding or
-	 * no_create_on_split, do not trust users coming back from splits;
-	 * with the exception of users coming back after a services
-	 * restart if the channel TS did not change.
-	 * Unfortunately, this kicks users who have been invited by a channel
-	 * operator, after a split.
-	 */
-	if (mc->mlock_on & CMODE_INVITE && !(flags & CA_INVITE) &&
-			(!me.bursting || mc->flags & MC_RECREATED) &&
-			(!(u->server->flags & SF_EOB) || (chan->nummembers - chan->numsvcmembers == 1)) &&
-			(!ircd->invex_mchar || !next_matching_ban(chan, u, ircd->invex_mchar, chan->bans.head)))
-	{
-		if (chan->nummembers - chan->numsvcmembers == 1)
-		{
-			mc->flags |= MC_INHABIT;
-			if (chan->numsvcmembers == 0)
-				join(chan->name, chansvs.nick);
-		}
-		if (!(chan->modes & CMODE_INVITE))
-			check_modes(mc, true);
-		modestack_flush_channel(chan);
-		try_kick(chansvs.me->me, chan, u, "Invite only channel");
-		hdata->cu = NULL;
-		return;
-	}
+	return_if_fail(mc != NULL);
 
-	/* A second user joined and was not kicked; we do not need
-	 * to stay on the channel artificially.
-	 * If there is only one user, stay in the channel to avoid
-	 * triggering autocycle-for-ops scripts and immediately
-	 * destroying channels with kick on split riding.
-	 */
-	if (mc->flags & MC_INHABIT && chan->nummembers - chan->numsvcmembers >= 2)
-	{
-		mc->flags &= ~MC_INHABIT;
-		if (!(mc->flags & MC_GUARD) && !(chan->flags & CHAN_LOG) && chanuser_find(chan, chansvs.me->me))
-			part(chan->name, chansvs.nick);
-	}
+	bool noop = mc->flags & MC_NOOP || (u->myuser != NULL &&
```

Comment:
This can all fit on one line. Would be nice to put parens around the first condition too.